File: backend/src/entity/JobLog.ts

import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn, CreateDateColumn } from "typeorm";
import { CronJob } from "./CronJob";
import { JobStatus } from "./CronJob";

@Entity()
export class JobLog {
  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column()
  jobId: string;

  @ManyToOne(() => CronJob, job => job.logs)
  @JoinColumn({ name: "jobId" })
  job: CronJob;

  @Column({
    type: "varchar",
    enum: ["idle", "running", "success", "failed", "paused"]
  })
  status: JobStatus;

  @Column({ type: "datetime" })
  startTime: Date;

  @Column({ type: "datetime", nullable: true })
  endTime: Date | null;

  @Column({ type: "float", nullable: true })
  duration: number | null;

  @Column({ type: "text" })
  output: string;

  @Column({ type: "text", nullable: true })
  error: string | null;

  @CreateDateColumn()
  createdAt: Date;
}

File: golang-backend/internal/models/job_log.go

package models

import (
	"time"

	"gorm.io/gorm"
)

type JobLog struct {
	ID        string    `json:"id" gorm:"primaryKey;type:varchar(36)"`
	JobID     string    `json:"jobId" gorm:"type:varchar(36);not null"`
	Status    JobStatus `json:"status" gorm:"type:varchar(10);not null"`
	StartTime time.Time `json:"startTime" gorm:"not null"`
	EndTime   time.Time `json:"endTime"`
	Duration  float64   `json:"duration"` // in seconds
	Output    string    `json:"output" gorm:"type:text"`
	Error     string    `json:"error" gorm:"type:text"`
	CreatedAt time.Time `json:"createdAt" gorm:"autoCreateTime"`
}

func (l *JobLog) BeforeCreate(tx *gorm.DB) (err error) {
	if l.ID == "" {
		l.ID = generateUUID()
	}
	return
}
